Duplicati/Library/Main/Operation/BackupHandler.cs

```diff
@@ -81,9 +81,11 @@ public static Snapshots.ISnapshotService GetSnapshot(string[] sources, Options o
             catch (Exception ex)
             {
                 if (options.SnapShotStrategy == Options.OptimizationStrategy.Required)
-                    throw;
+                    throw new Exception("Failed to create a snapshot");

```

Comment:
We should try to [avoid throwing](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/using-standard-exception-types#exception-and-systemexception) the base `System.Exception` type.  Doing so can make the exception more difficult to handle downstream.  If none of the standard exception types are applicable, we can add a custom type in the `CustomExceptions` class.  We can then throw the custom exception and provide `ex` as the [inner exception](https://docs.microsoft.com/en-us/dotnet/api/system.exception.-ctor?view=netframework-4.8#System_Exception__ctor_System_String_System_Exception_).

Reply:
Ok, updated to use `UserInformationException` class.  Also changed it to use the proper string from `Strings.cs`.